packages/kubernetes-1.22/credential-provider-config-yaml

```diff
@@ -15,9 +15,9 @@ providers:
     env:
       - name: HOME
         value: /root
-{{#if settings.aws.profile}}
+{{#if @root.settings.aws.profile}}
```

Comment:
Out of curiosity, why was the `@root.` addition needed?

Reply:
Yeah, that was the tricky, key part of this.

Because this is inside an [{{#each}}](https://github.com/bottlerocket-os/bottlerocket/blob/f825b8186faabe4283a06ead96b2bfd4123bd0da/packages/kubernetes-1.24/credential-provider-config-yaml#L5) block, that means handlebars iterates through "each" of the `settings.kubernetes.credential-providers`  values in that collection.

What's not immediately obvious is that it also means you are within the context of those iterated values. So that's why on the next line you can do:

```
{{#if this.enabled}}
```

instead of some sort of indexing operation into the collection. But that's also where the subtle gotcha is.

It meant that later on with the line:

```
{{#if settings.aws.profile}}
```

it was actually looking at `settings.kubernetes.credential-providers[x].settings.aws.profile`. Since we're inside the `each` block, we actually need to break out of that local context to get to the real `settings.aws.profile`.

Handlebars allows you to do this in two different ways - either relationally (`../../aws.profile`) or by explicitly going from the anchor like what has been done here. This looks a lot cleaner, and probably safer, so I went with the `@root` route.